exe/floe

```diff
@@ -4,29 +4,37 @@
 require "floe"
 require "optimist"
 
+supported_docker_runners = {
+  "docker"     => Floe::Workflow::Runner::Docker, # first is the default
+  "podman"     => Floe::Workflow::Runner::Podman,
+  "kubernetes" => Floe::Workflow::Runner::Kubernetes
+}
+
 opts = Optimist.options do
   version("v#{Floe::VERSION}\n")
+
   opt :workflow, "Path to your workflow json",                 :type => :string, :required => true
   opt :input, "JSON payload to input to the workflow",         :default => '{}'
   opt :credentials, "JSON payload with credentials",           :default => "{}"
-  opt :docker_runner, "Type of runner for docker images",      :default => "docker"
-  opt :docker_runner_options, "Options to pass to the runner", :type => :strings
+  opt :docker_runner, "Type of runner for docker images",      :type => :string, :short => 'r'
+  opt :docker_runner_options, "Options to pass to the runner", :type => :strings, :short => 'o'
+
+  supported_docker_runners.each_key do |runner|
+    opt runner.to_sym, "Use #{runner} to run images (short for --docker_runner=#{runner})", :type => :boolean
+  end
 end
 
-Optimist.die(:docker_runner, "must be one of #{Floe::Workflow::Runner::TYPES.join(", ")}") unless Floe::Workflow::Runner::TYPES.include?(opts[:docker_runner])
+# if they specified a docker_runner, and it is unknown, complain
+Optimist.die(:docker_runner, "must be one of #{supported_docker_runners.join(", ")}") if opts[:docker_runner] && !supported_docker_runners.key?(opts[:docker_runner])
 
 require "logger"
 Floe.logger = Logger.new($stdout)
 
-runner_klass = case opts[:docker_runner]
-               when "docker"
-                 Floe::Workflow::Runner::Docker
-               when "podman"
-                 Floe::Workflow::Runner::Podman
-               when "kubernetes"
-                 Floe::Workflow::Runner::Kubernetes
-               end
-
+# use docker_runner parameter, shortcut, or default
+runner_klass =
+  supported_docker_runners[opts[:docker_runner]] ||
+  supported_docker_runners.detect { |type, _klass| opts[type.to_sym] }&.last ||
+  supported_docker_runners.values.first
```

Comment:
I like the conversion from case to hash.

But the `detect` and code is not as clear

We could also do:

```ruby
opts[:docker_runner] ||= opts[:docker] && "docker" || opts[:kubernetes] && "kubernetes || opts[:podman] && "podman" || support_docker_runners.values.first
```

or what we originally had:

```ruby
opts[:docker_runner] ||= "docker" if opts[:docker]
opts[:docker_runner] ||= "podman" if opts[:podman]
opts[:docker_runner] ||= "kubernetes" if opts[:kubernetes]
opts[:docker_runner] ||= support_docker_runners.values.first
```

Reply:
also tried:

```ruby
SUPPORTED_DOCKER_RUNNERS.each_key do |runner_name|
  opts[:docker_runner] ||= runner_name if opts[runner_name.to_sym]
end
```

Keep going back to the duplicate code version. It takes no braincells to read and is the same number of lines.